docs/resources/dms_rocketmq_instance.md

```diff
@@ -0,0 +1,178 @@
+---
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# huaweicloud_dms_rocketmq_instance
+
+  Manage DMS RocketMQ instance resources within HuaweiCloud.
+
+## Example Usage
+
+```HCL
+resource "huaweicloud_dms_rocketmq_instance" "test" {
+  name                = "rocketmq_name_test"
+  description         = "this is a rocketmq instance"
+  engine              = "reliability"
+  engine_version      = "4.8.0"
+  total_storage_space = 10
+  vpc_id              = "4579f2f5-cbe8-425a-8f32-53dcb9d9053a"
+  subnet_id           = "4579f2f5-cbe8-425a-8f32-53dcb9d9053a"
+  security_group_id   = "4579f2f5-cbe8-425a-8f32-53dcb9d9053a"
+  available_zones     = ["cn-north-4a", "cn-north-4b"]
+  product_id          = "c6.4u8g.cluster"
+  storage_spec_code   = "dms.physical.storage.high.v2"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Instance name.
+  An instance name starts with a letter, consists of 4 to 64 characters, and can contain only letters,
+  digits, underscores (_), and hyphens (-).
+
+* `engine` - (Required, String, ForceNew) Message engine. Value: reliability.
+  Changing this parameter will create a new resource.
+
+* `engine_version` - (Required, String, ForceNew) Version of the message engine. Value: 4.8.0.
+  Changing this parameter will create a new resource.
+
+* `total_storage_space` - (Required, Int, ForceNew) Total storage space.
```

Comment:
add the value range description as there is limit in the code

Reply:
done